nntrainer/tensor/cache_pool.cpp

```diff
@@ -11,51 +11,81 @@
  *
  */
 
-#include "memory_pool.h"
+#include "cache_pool.h"
+
 #include <limits>
 #include <numeric>
 #include <stdexcept>
 #include <vector>
 
-#include <cache_pool.h>
 #include <nntrainer_error.h>
 #include <nntrainer_log.h>
 #include <profiler.h>
 
 namespace nntrainer {
 
-void CacheElem::swapIn() {
-  std::lock_guard<std::mutex> lock(device_mutex);
-  void *buf = device->getBuffer(offset, length);
-  mem_data->setAddr((float *)buf);
-  mem_data->setValid(true);
-  active = true;
+namespace {
+
+inline const CachePolicy
+convertTensorLifespanToCachePolicy(const TensorLifespan lifespan) {
+  CachePolicy policy;
+
+  switch (lifespan) {
+  case TensorLifespan::UNMANAGED:
+    policy = CachePolicy::ALWAYS_SYNCED;
+    break;
+  case TensorLifespan::FORWARD_FUNC_LIFESPAN:
+    policy = CachePolicy::ALWAYS_SYNCED;
+    break;
+  case TensorLifespan::CALC_DERIV_LIFESPAN:
+    policy = CachePolicy::TEMPORAL;
+    break;
+  case TensorLifespan::CALC_GRAD_LIFESPAN:
+    policy = CachePolicy::TEMPORAL;
+    break;
+  case TensorLifespan::CALC_GRAD_DERIV_LIFESPAN:
+    policy = CachePolicy::TEMPORAL;
+    break;
+  case TensorLifespan::FORWARD_GRAD_LIFESPAN:
+    policy = CachePolicy::ITERATION_CONSIST;
+    break;
+  case TensorLifespan::FORWARD_DERIV_LIFESPAN:
+    policy = CachePolicy::ALWAYS_SYNCED;
+    break;
+  case TensorLifespan::ITERATION_LIFESPAN:
+    policy = CachePolicy::ITERATION_CONSIST;
+    break;
+  case TensorLifespan::EPOCH_LIFESPAN:
+    policy = CachePolicy::ITERATION_CONSIST;
+    break;
+  case TensorLifespan::MAX_LIFESPAN:
+    policy = CachePolicy::ALWAYS_SYNCED;
+    break;
+  default:
+    policy = CachePolicy::ALWAYS_SYNCED;
+    break;
+  }
 
-  std::string msg("CacheElem(");
-  msg += device->getDevicePath() + ") #" + std::to_string(id);
-  PROFILE_MEM_ALLOC(buf, length, msg);
+  return policy;
 }
 
-void CacheElem::swapOut() {
-  std::lock_guard<std::mutex> lock(device_mutex);
-  void *buf = (void *)mem_data->getAddr();
-  device->putBuffer(buf);
-  mem_data->setAddr(nullptr);
-  mem_data->setValid(false);
-  active = false;
+std::atomic_int pool_id = 0;
 
-  PROFILE_MEM_DEALLOC(buf);
-}
+} // namespace
 
-CachePool::CachePool(const std::string &name) :
-  swap_device(std::make_shared<SwapDevice>(name + std::to_string(getpid()))) {}
+CachePool::CachePool(const std::string &n) :
```

Comment:
Does the n is for path of swap device? if it is could you add the comment in ```CachePool``` constructor in header?

Reply:
thanks. it added.